sdks/java/core/src/test/java/org/apache/beam/sdk/metrics/MetricsTest.java

```diff
@@ -95,4 +107,54 @@ public void counterToCell() {
     counter.dec();
     assertThat(cell.getCumulative(), CoreMatchers.equalTo(42L));
   }
+
+  @Category({RunnableOnService.class, UsesMetrics.class})
+  @Test
+  public void metricsE2E() {
```

Comment:
This tests two things:

The ability to report metrics on the given runner *and* the ability to get them from the pipeline.